.github/workflows/jekyll-gh-pages.yml

```diff
@@ -35,19 +40,29 @@ jobs:
           cache-version: 0 # Increment this number if you need to re-download cached gems
       - name: Setup Pages
         uses: actions/configure-pages@v3
-      - name: Install rake
-        run: gem install rake
-      - name: Update Bundler and Setup Jekyll
-        run: bundle install && bundle update --bundler
-      - name: Setup Go
-        uses: actions/setup-go@v5
-      - name: Setup ASCIIToSVG
-        run: go install github.com/asciitosvg/asciitosvg/cmd/a2s@latest
-      - name: Download Bootstrap 5 SCSS Sources
-        run: ./bootstrap_setup.sh
+      
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          driver-opts: network=host
+
+      - name: Build Docker image
+        uses: docker/build-push-action@v6
+        with:
+          context: .
+          push: true
+          tags: localhost:5000/kroxy-jekyll:latest
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
       - name: Build with Jekyll
         # Outputs to the './_site' directory by default
-        run: bundle exec jekyll build --baseurl "${{ steps.pages.outputs.base_path }}"
+        run: |
+          docker run \
+            --rm \
+            -u "$(id -u):$(id -g)" \
+            -v "$(pwd):/site" \
+            localhost:5000/kroxy-jekyll:latest \
+            bash -c 'eval "$(rbenv init -)" && cp -r /css/_sass/bootstrap /site/_sass/ && bundle exec jekyll build --baseurl "${{ steps.pages.outputs.base_path }}"'
```

Comment:
the `--baseurl "${{ steps.pages.outputs.base_path }}` isn't actually doing anything because we don't have an `id: step` on the earlier configure pages step.

It's kind of moot since the pages are generally pretty broken if you deploy them on a fork, the docs links don't appear to honour base_url and some links like the quickstarts in the top nav go straight to kroxylicious.io which I think is because it's configured into _config.yml as the url. Needs a whole other PR to try and get it set up to work on a fork better.

Reply:
(existing issues with running on forks)

Reply:
What do you mean by "deploy on a fork"?

Reply:
I have my fork set up to deploy a branch to github pages

https://robobario.github.io/kroxylicious.github.io/

because of the way the job is configured with site.url configured to `kroylicious.io` and it not being aware of the extra `kroxylicious.github.io` it looks wrong.

with a couple of changes we could make it work neatly on a fork, it helps when testing changes to the github workflows

Reply:
edit: sorry, I mean it deploys the same way as on this repo. I push to main on my fork and the action deploys to pages.